gateway-js/src/index.ts

```diff
@@ -255,87 +288,133 @@ export class ApolloGateway implements GraphQLService {
       ...config,
     };
 
+    this.logger = this.initLogger();
+    this.queryPlanStore = this.initQueryPlanStore();
+    this.fetcher = config?.fetcher || getDefaultGcsFetcher();
+
+    // set up experimental observability callbacks and config settings
+    this.experimental_didResolveQueryPlan =
+      config?.experimental_didResolveQueryPlan;
+    this.experimental_didFailComposition =
+      config?.experimental_didFailComposition;
+    this.experimental_didFailComposition =
+      config?.experimental_didFailComposition;
+    this.experimental_didUpdateComposition =
+      config?.experimental_didUpdateComposition;
+    this.experimental_approximateQueryPlanStoreMiB =
```

Comment:
This looks like it happens too late, since you moved up the call to initQueryPlanStore. That said, you can probably just remove the field from ApolloGateway and pass `config?.experimental_approximateQueryPlanStoreMiB` directly to initQueryPlanStore?

Reply:
Good catch, agree with your suggestion too.
00a5eca